packages/styles/scss/components/data-table/_data-table-action.scss

```diff
@@ -0,0 +1,586 @@
+    .#{$prefix}--search-magnifier-icon {
+    background-color: $layer-disabled;
+    cursor: not-allowed;
+    transition: background-color none;
+  }
+
+  .#{$prefix}--toolbar-search-container-disabled .#{$prefix}--search-input {
+    cursor: not-allowed;
+  }
+
+  .#{$prefix}--toolbar-search-container-expandable.#{$prefix}--search
+    .#{$prefix}--label {
+    visibility: hidden;
+  }
+
+  .#{$prefix}--toolbar-search-container-expandable.#{$prefix}--search
+    .#{$prefix}--search-close {
+    width: $spacing-09;
+    height: $spacing-09;
+
+    &::before {
+      top: rem(2px);
+      height: calc(100% - #{rem(4px)});
+      background-color: $field-hover;
+    }
+  }
+
+  .#{$prefix}--toolbar-search-container-expandable.#{$prefix}--search
+    .#{$prefix}--search-close:focus::before {
+    background-color: $focus;
+  }
+
+  //-------------------------------------------------
+  //ACTIVE SEARCH - DEFAULT TOOLBAR
+  //-------------------------------------------------
+
+  .#{$prefix}--toolbar-search-container-active.#{$prefix}--search {
+    width: 100%;
+  }
+
+  .#{$prefix}--toolbar-search-container-active .#{$prefix}--search-input {
+    opacity: 1;
+  }
+
+  .#{$prefix}--toolbar-search-container-active .#{$prefix}--label,
+  .#{$prefix}--toolbar-search-container-active .#{$prefix}--search-input {
+    padding: 0 $spacing-09;
+    cursor: text;
+  }
+
+  .#{$prefix}--toolbar-search-container-active
+    .#{$prefix}--search-input:focus
+    + .#{$prefix}--search-close {
+    border: none;
+    box-shadow: none;
+    outline: none;
+  }
+
+  .#{$prefix}--toolbar-search-container-active
+    .#{$prefix}--search-input:not(:placeholder-shown) {
+    border: none;
+    background-color: $field-hover;
+  }
+
+  .#{$prefix}--toolbar-search-container-active
+    .#{$prefix}--search-magnifier-icon:focus,
+  .#{$prefix}--toolbar-search-container-active
+    .#{$prefix}--search-magnifier-icon:active,
+  .#{$prefix}--toolbar-search-container-active
+    .#{$prefix}--search-magnifier-icon:hover {
+    border: none;
+    background-color: transparent;
+    outline: none;
+  }
+
+  //-------------------------------------------------
+  //SEARCH CLOSE BUTTON
+  //-------------------------------------------------
+  .#{$prefix}--toolbar-search-container-persistent .#{$prefix}--search-close,
+  .#{$prefix}--toolbar-search-container-persistent
+    .#{$prefix}--search-close:hover,
+  .#{$prefix}--toolbar-search-container-active .#{$prefix}--search-close,
+  .#{$prefix}--toolbar-search-container-active .#{$prefix}--search-close:hover {
+    border: none;
+    background-color: transparent;
+  }
+
+  .#{$prefix}--toolbar-search-container-persistent
+    .#{$prefix}--search-close::before {
+    display: none;
+  }
+
+  //-------------------------------------------------
+  //TOOLBAR BUTTONS
+  //-------------------------------------------------
+  .#{$prefix}--overflow-menu.#{$prefix}--toolbar-action {
```

Comment:
Ah, didn't think about this but will we need specific component dependencies in for data table before we are able to land data table?

Reply:
Yeah, we'll need those before we add DataTable to the new `/carbon-react` storybook. I don't think we need them to land this port over to sass modules though?